components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SessionDataStore.java

```diff
@@ -122,39 +131,71 @@
     private static final String INFORMIX_DATABASE = "Informix";
 
     private static final int DEFAULT_DELETE_LIMIT = 50000;
-    private static int maxPoolSize = 100;
+    public static final String SESSION_STORE_TABLE = "IDN_AUTH_SESSION_STORE";
+    public static final String TEMP_SESSION_STORE_TABLE = "IDN_AUTH_SESSION_STORE_TEMP";
+    private static int maxSessionDataPoolSize = 20;
+    private static int maxTempSessionDataPoolSize = 100;
+    private static int maxDeleteTempDataPoolSize = 50;
     private static BlockingDeque<SessionContextDO> sessionContextQueue = new LinkedBlockingDeque();
+    private static BlockingDeque<SessionContextDO> tempSessionContextQueue = new LinkedBlockingDeque();
+    private static BlockingDeque<SessionContextDO> tempSessionContextDeleteQueue = new LinkedBlockingDeque();
     private static volatile SessionDataStore instance;
     private boolean enablePersist;
     private String sqlInsertSTORE;
+    private String sqlInsertTempSTORE;
     private String sqlInsertDELETE;
+    private String sqlInsertTempDELETE;
     private String sqlDeleteSTORETask;
     private String sqlDeleteDELETETask;
     private String sqlSelect;
     private String sqlDeleteExpiredDataTask;
     private int deleteChunkSize = DEFAULT_DELETE_LIMIT;
     private boolean sessionDataCleanupEnabled = true;
     private boolean operationDataCleanupEnabled = false;
+    private List<String> temporaryCache;
+
 
     static {
         try {
-            String maxPoolSizeConfigValue = IdentityUtil.getProperty("JDBCPersistenceManager.SessionDataPersist.PoolSize");
-            if (StringUtils.isNotBlank(maxPoolSizeConfigValue)) {
-                maxPoolSize = Integer.parseInt(maxPoolSizeConfigValue);
+            String maxPoolSizeValue = IdentityUtil.getProperty("JDBCPersistenceManager.SessionDataPersist.PoolSize");
+            String maxTempDataPoolSizeValue = IdentityUtil.getProperty("JDBCPersistenceManager.SessionDataPersist.TempDataPoolSize");
+            String maxTempDataDeletePoolSizeValue = IdentityUtil.getProperty("JDBCPersistenceManager.SessionDataPersist.TempDataDeletePoolSize");
+            if (StringUtils.isNotBlank(maxPoolSizeValue)) {
+                maxSessionDataPoolSize = Integer.parseInt(maxPoolSizeValue);
+            }
+            if (StringUtils.isNotBlank(maxTempDataPoolSizeValue)) {
+                maxTempSessionDataPoolSize = Integer.parseInt(maxTempDataPoolSizeValue);
+            }
+            if (StringUtils.isNotBlank(maxTempDataDeletePoolSizeValue)) {
+                maxDeleteTempDataPoolSize = Integer.parseInt(maxTempDataDeletePoolSizeValue);
             }
         } catch (NumberFormatException e) {
             if (log.isDebugEnabled()) {
```

Comment:
We just have only a single catch for parsing 3 strings to Integers. This warning log doesn't say exactly which string failed to parse.